sdk/openai/azure-ai-openai/src/main/java/com/azure/ai/openai/OpenAIClientBuilder.java

```diff
@@ -182,23 +181,22 @@ public OpenAIClientBuilder credential(TokenCredential tokenCredential) {
     @Generated private AzureKeyCredential azureKeyCredential;
 
     /** {@inheritDoc}. */
-    @Generated
     @Override
     public OpenAIClientBuilder credential(AzureKeyCredential azureKeyCredential) {
-        this.azureKeyCredential = azureKeyCredential;
-        return this;
+        return this.credential((KeyCredential) azureKeyCredential);
```

Comment:
Why do we want to keep this overload? We should delete it as we are still in beta.

Reply:
talked offline. Will keep this method until code-gen fixes the issue. (We can't remove `AzureKeyCredentialTrait,`